src/mqt/predictor/ml/helper.py

```diff
@@ -23,36 +23,41 @@
 
 def qcompile(
     qc: QuantumCircuit,
+    devices: list[str] | str | None = None,
```

Comment:
```suggestion
    devices: list[str] | None = None,
```
based on the comments in the documentation, the above should be enough.
You might want to take this one step further and use `Iterable[str]` (from `collections.abc`) instead of `list[str]`. 
Interfaces should typically be as flexible as possible in the parameters they take and as narrow as possible in the types they return.
This comment holds for all the changes in this PR and some of the rest of the predictor code.

Reply:
I implemented the suggestion from above. While I do agree with your statement, I do not have the headspace to implement the further improvements in this PR right now.

Reply:
That makes sense 👍🏼